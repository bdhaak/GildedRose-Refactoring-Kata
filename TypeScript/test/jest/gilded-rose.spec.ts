import { Item, GildedRose } from '@/gilded-rose';
import { describe, expect, test } from "@jest/globals";
import {verify, verifyAll, verifyAsJson} from "approvals/lib/Providers/Jest/JestApprovals";
import {verifyAllCombinations3} from "approvals/lib/Providers/Jest/CombinationApprovals";

describe('Gilded Rose', () => {
  
  // Original. Can be fixed easily. 
  it('should foo', () => {
    const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
    const items = gildedRose.updateQuality();
    // expect(items[0].name).toBe('fixme');
    expect(items[0].name).toBe('foo');
  });

  // Step 1. Using approvals 
  it('should update quality', () => {
    const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
    const items = gildedRose.updateQuality();
    //expect(items[0].name).toBe('foo');
    verify(items[0].name);
  });

  // Step 2. Using approvals, json
  it('should update quality with json approvals', () => {
    const gildedRose = new GildedRose([new Item('foo', 0, 0)]);
    const items = gildedRose.updateQuality();
    verifyAsJson(items[0]);
  });

  // Step 3. Using combination approvals
  it('should update quality with combination approvals', () => {  
    verifyAllCombinations3( updateQuality,
        ['Foo',
        'Aged Brie', 
        'Sulfuras, Hand of Ragnaros', 
        'Backstage passes to a TAFKAL80ETC concert'], 
        [-1, 0, 5, 6, 10, 11], 
        [0, 1, 49, 50, 51] );  
    
    });
});

function updateQuality(name: string, sellIn: number, quality: number) : string{
  const gildedRose = new GildedRose([new Item(name, sellIn, quality)]);
  const items = gildedRose.updateQuality();
  const qualityItem = items[0];
  return `${qualityItem.name}, ${qualityItem.sellIn}, ${qualityItem.quality}`;
}

