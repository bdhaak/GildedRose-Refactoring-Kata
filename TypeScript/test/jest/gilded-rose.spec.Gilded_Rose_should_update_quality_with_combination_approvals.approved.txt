[Foo,-1,0] => Foo, -2, 0
[Foo,-1,1] => Foo, -2, 0
[Foo,-1,49] => Foo, -2, 47
[Foo,-1,50] => Foo, -2, 48
[Foo,-1,51] => Foo, -2, 49
[Foo,0,0] => Foo, -1, 0
[Foo,0,1] => Foo, -1, 0
[Foo,0,49] => Foo, -1, 47
[Foo,0,50] => Foo, -1, 48
[Foo,0,51] => Foo, -1, 49
[Foo,5,0] => Foo, 4, 0
[Foo,5,1] => Foo, 4, 0
[Foo,5,49] => Foo, 4, 48
[Foo,5,50] => Foo, 4, 49
[Foo,5,51] => Foo, 4, 50
[Foo,6,0] => Foo, 5, 0
[Foo,6,1] => Foo, 5, 0
[Foo,6,49] => Foo, 5, 48
[Foo,6,50] => Foo, 5, 49
[Foo,6,51] => Foo, 5, 50
[Foo,10,0] => Foo, 9, 0
[Foo,10,1] => Foo, 9, 0
[Foo,10,49] => Foo, 9, 48
[Foo,10,50] => Foo, 9, 49
[Foo,10,51] => Foo, 9, 50
[Foo,11,0] => Foo, 10, 0
[Foo,11,1] => Foo, 10, 0
[Foo,11,49] => Foo, 10, 48
[Foo,11,50] => Foo, 10, 49
[Foo,11,51] => Foo, 10, 50
[Aged Brie,-1,0] => Aged Brie, -2, 2
[Aged Brie,-1,1] => Aged Brie, -2, 3
[Aged Brie,-1,49] => Aged Brie, -2, 50
[Aged Brie,-1,50] => Aged Brie, -2, 50
[Aged Brie,-1,51] => Aged Brie, -2, 51
[Aged Brie,0,0] => Aged Brie, -1, 2
[Aged Brie,0,1] => Aged Brie, -1, 3
[Aged Brie,0,49] => Aged Brie, -1, 50
[Aged Brie,0,50] => Aged Brie, -1, 50
[Aged Brie,0,51] => Aged Brie, -1, 51
[Aged Brie,5,0] => Aged Brie, 4, 1
[Aged Brie,5,1] => Aged Brie, 4, 2
[Aged Brie,5,49] => Aged Brie, 4, 50
[Aged Brie,5,50] => Aged Brie, 4, 50
[Aged Brie,5,51] => Aged Brie, 4, 51
[Aged Brie,6,0] => Aged Brie, 5, 1
[Aged Brie,6,1] => Aged Brie, 5, 2
[Aged Brie,6,49] => Aged Brie, 5, 50
[Aged Brie,6,50] => Aged Brie, 5, 50
[Aged Brie,6,51] => Aged Brie, 5, 51
[Aged Brie,10,0] => Aged Brie, 9, 1
[Aged Brie,10,1] => Aged Brie, 9, 2
[Aged Brie,10,49] => Aged Brie, 9, 50
[Aged Brie,10,50] => Aged Brie, 9, 50
[Aged Brie,10,51] => Aged Brie, 9, 51
[Aged Brie,11,0] => Aged Brie, 10, 1
[Aged Brie,11,1] => Aged Brie, 10, 2
[Aged Brie,11,49] => Aged Brie, 10, 50
[Aged Brie,11,50] => Aged Brie, 10, 50
[Aged Brie,11,51] => Aged Brie, 10, 51
[Sulfuras, Hand of Ragnaros,-1,0] => Sulfuras, Hand of Ragnaros, -1, 0
[Sulfuras, Hand of Ragnaros,-1,1] => Sulfuras, Hand of Ragnaros, -1, 1
[Sulfuras, Hand of Ragnaros,-1,49] => Sulfuras, Hand of Ragnaros, -1, 49
[Sulfuras, Hand of Ragnaros,-1,50] => Sulfuras, Hand of Ragnaros, -1, 50
[Sulfuras, Hand of Ragnaros,-1,51] => Sulfuras, Hand of Ragnaros, -1, 51
[Sulfuras, Hand of Ragnaros,0,0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros,0,1] => Sulfuras, Hand of Ragnaros, 0, 1
[Sulfuras, Hand of Ragnaros,0,49] => Sulfuras, Hand of Ragnaros, 0, 49
[Sulfuras, Hand of Ragnaros,0,50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros,0,51] => Sulfuras, Hand of Ragnaros, 0, 51
[Sulfuras, Hand of Ragnaros,5,0] => Sulfuras, Hand of Ragnaros, 5, 0
[Sulfuras, Hand of Ragnaros,5,1] => Sulfuras, Hand of Ragnaros, 5, 1
[Sulfuras, Hand of Ragnaros,5,49] => Sulfuras, Hand of Ragnaros, 5, 49
[Sulfuras, Hand of Ragnaros,5,50] => Sulfuras, Hand of Ragnaros, 5, 50
[Sulfuras, Hand of Ragnaros,5,51] => Sulfuras, Hand of Ragnaros, 5, 51
[Sulfuras, Hand of Ragnaros,6,0] => Sulfuras, Hand of Ragnaros, 6, 0
[Sulfuras, Hand of Ragnaros,6,1] => Sulfuras, Hand of Ragnaros, 6, 1
[Sulfuras, Hand of Ragnaros,6,49] => Sulfuras, Hand of Ragnaros, 6, 49
[Sulfuras, Hand of Ragnaros,6,50] => Sulfuras, Hand of Ragnaros, 6, 50
[Sulfuras, Hand of Ragnaros,6,51] => Sulfuras, Hand of Ragnaros, 6, 51
[Sulfuras, Hand of Ragnaros,10,0] => Sulfuras, Hand of Ragnaros, 10, 0
[Sulfuras, Hand of Ragnaros,10,1] => Sulfuras, Hand of Ragnaros, 10, 1
[Sulfuras, Hand of Ragnaros,10,49] => Sulfuras, Hand of Ragnaros, 10, 49
[Sulfuras, Hand of Ragnaros,10,50] => Sulfuras, Hand of Ragnaros, 10, 50
[Sulfuras, Hand of Ragnaros,10,51] => Sulfuras, Hand of Ragnaros, 10, 51
[Sulfuras, Hand of Ragnaros,11,0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros,11,1] => Sulfuras, Hand of Ragnaros, 11, 1
[Sulfuras, Hand of Ragnaros,11,49] => Sulfuras, Hand of Ragnaros, 11, 49
[Sulfuras, Hand of Ragnaros,11,50] => Sulfuras, Hand of Ragnaros, 11, 50
[Sulfuras, Hand of Ragnaros,11,51] => Sulfuras, Hand of Ragnaros, 11, 51
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,49] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,51] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,0,0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,49] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,51] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,5,0] => Backstage passes to a TAFKAL80ETC concert, 4, 3
[Backstage passes to a TAFKAL80ETC concert,5,1] => Backstage passes to a TAFKAL80ETC concert, 4, 4
[Backstage passes to a TAFKAL80ETC concert,5,49] => Backstage passes to a TAFKAL80ETC concert, 4, 50
[Backstage passes to a TAFKAL80ETC concert,5,50] => Backstage passes to a TAFKAL80ETC concert, 4, 50
[Backstage passes to a TAFKAL80ETC concert,5,51] => Backstage passes to a TAFKAL80ETC concert, 4, 51
[Backstage passes to a TAFKAL80ETC concert,6,0] => Backstage passes to a TAFKAL80ETC concert, 5, 2
[Backstage passes to a TAFKAL80ETC concert,6,1] => Backstage passes to a TAFKAL80ETC concert, 5, 3
[Backstage passes to a TAFKAL80ETC concert,6,49] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert,6,50] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert,6,51] => Backstage passes to a TAFKAL80ETC concert, 5, 51
[Backstage passes to a TAFKAL80ETC concert,10,0] => Backstage passes to a TAFKAL80ETC concert, 9, 2
[Backstage passes to a TAFKAL80ETC concert,10,1] => Backstage passes to a TAFKAL80ETC concert, 9, 3
[Backstage passes to a TAFKAL80ETC concert,10,49] => Backstage passes to a TAFKAL80ETC concert, 9, 50
[Backstage passes to a TAFKAL80ETC concert,10,50] => Backstage passes to a TAFKAL80ETC concert, 9, 50
[Backstage passes to a TAFKAL80ETC concert,10,51] => Backstage passes to a TAFKAL80ETC concert, 9, 51
[Backstage passes to a TAFKAL80ETC concert,11,0] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert,11,1] => Backstage passes to a TAFKAL80ETC concert, 10, 2
[Backstage passes to a TAFKAL80ETC concert,11,49] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert,11,50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert,11,51] => Backstage passes to a TAFKAL80ETC concert, 10, 51
